test/functional/Java14andUp/playlist.xml

```diff
@@ -23,28 +23,51 @@
 -->
 
 <playlist xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../../TestConfig/playlist.xsd">
-      <test>
-          <testCaseName>Jep359Tests</testCaseName>
-          <variations>
-                  <variation>NoOptions</variation>
-          </variations>
-          <command>$(JAVA_COMMAND) $(JVM_OPTIONS) \
-                  --enable-preview \
-                  -cp $(Q)$(LIB_DIR)$(D)asm-7.3.jar$(P)$(RESOURCES_DIR)$(P)$(TESTNG)$(P)$(TEST_RESROOT)$(D)GeneralTest.jar$(Q) \
-                  org.testng.TestNG -d $(REPORTDIR) $(Q)$(TEST_RESROOT)$(D)testng.xml$(Q) -testnames Jep359Tests \
-                  -groups $(TEST_GROUP) \
-                  -excludegroups $(DEFAULT_EXCLUDE); \
-                  $(TEST_STATUS)
-          </command>
-          <levels>
-                  <level>sanity</level>
-          </levels>
-          <groups>
-                  <group>functional</group>
-          </groups>
-          <!-- Run for Java 14 only since this is a preview feature. -->
-          <subsets>
-                  <subset>14</subset>
-          </subsets>
-        </test>
+    <test>
+        <testCaseName>Jep359Tests</testCaseName>
+        <variations>
+            <variation>NoOptions</variation>
+        </variations>
+        <command>$(JAVA_COMMAND) $(JVM_OPTIONS) \
+            --enable-preview \
+            -cp $(Q)$(LIB_DIR)$(D)asm-7.3.jar$(P)$(RESOURCES_DIR)$(P)$(TESTNG)$(P)$(TEST_RESROOT)$(D)GeneralTest.jar$(Q) \
+            org.testng.TestNG -d $(REPORTDIR) $(Q)$(TEST_RESROOT)$(D)testng.xml$(Q) -testnames Jep359Tests \
+            -groups $(TEST_GROUP) \
+            -excludegroups $(DEFAULT_EXCLUDE); \
+            $(TEST_STATUS)
+        </command>
+        <levels>
+            <level>sanity</level>
+        </levels>
+        <groups>
+            <group>functional</group>
+        </groups>
+        <!-- Run for Java 14 only since this is a preview feature. -->
+        <subsets>
+            <subset>14</subset>
+        </subsets>
+    </test>
+    <test>
+        <testCaseName>ThreadInterruptImplTest</testCaseName>
+        <variations>
+            <variation>NoOptions</variation>
+        </variations>
+        <command>$(JAVA_COMMAND) $(JVM_OPTIONS) \
+            --enable-preview \
```

Comment:
Is `--enable-preview` needed for this test?

Reply:
Without --enable-preview here I am getting `java.lang.UnsupportedClassVersionError: Preview features are not enabled for org/openj9/test/java/lang/Test_Class` when I try to run ThreadInterruptImplTest.

Is there a preferable way to set this up @llxia? Note Test_Class is part of Jep359Tests and is not related to ThreadInterruptImplTest.

Reply:
Will we remove `--enable-preview` option in the future? If so, you can move it in variation ` <variation>--enable-preview</variation>`, so it is more visible.

Reply:
So now I am curious why org/openj9/test/java/lang/Test_Class is involved when we're running tests from org/openj9/test/java/lang/Test_Thread. Is it just some side affect of testng? Do you have the stack trace?

Reply:
```
Exception in thread "main" java.lang.UnsupportedClassVersionError: Preview features are not enabled for org/openj9/test/java/lang/Test_Class (class file version 58.65535). Try running with '--enable-preview'
        at java.base/java.lang.ClassLoader.defineClass1(Native Method)
        at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1017)
        at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:151)
        at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:821)
        at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:719)
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:642)
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:600)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
        at org.testng.internal.ClassHelper.forName(ClassHelper.java:117)
        at org.testng.xml.XmlClass.loadClass(XmlClass.java:74)
        at org.testng.xml.XmlClass.init(XmlClass.java:69)
        at org.testng.xml.XmlClass.<init>(XmlClass.java:55)
        at org.testng.xml.TestNGContentHandler.startElement(TestNGContentHandler.java:575)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.startElement(AbstractSAXParser.java:518)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractXMLDocumentParser.emptyElement(AbstractXMLDocumentParser.java:183)
        at java.xml/com.sun.org.apache.xerces.internal.impl.dtd.XMLDTDValidator.emptyElement(XMLDTDValidator.java:752)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.scanStartElement(XMLNSDocumentScannerImpl.java:351)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl$FragmentContentDriver.next(XMLDocumentFragmentScannerImpl.java:2725)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:605)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next(XMLNSDocumentScannerImpl.java:112)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:541)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:888)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:824)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(AbstractSAXParser.java:1224)
        at java.xml/com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(SAXParserImpl.java:635)
        at java.xml/com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl.parse(SAXParserImpl.java:324)
        at java.xml/javax.xml.parsers.SAXParser.parse(SAXParser.java:197)
        at org.testng.xml.XMLParser.parse(XMLParser.java:38)
        at org.testng.xml.SuiteXmlParser.parse(SuiteXmlParser.java:16)
        at org.testng.xml.SuiteXmlParser.parse(SuiteXmlParser.java:9)
        at org.testng.xml.Parser.parse(Parser.java:152)
        at org.testng.xml.Parser.parse(Parser.java:233)
        at org.testng.TestNG.parseSuite(TestNG.java:295)
        at org.testng.TestNG.initializeSuitesAndJarFile(TestNG.java:348)
        at org.testng.TestNG.initializeEverything(TestNG.java:995)
        at org.testng.TestNG.run(TestNG.java:1009)
        at org.testng.TestNG.privateMain(TestNG.java:1354)
        at org.testng.TestNG.main(TestNG.java:1323)
```

Reply:
I think the issue is that TestNG includes all tests from the suite on the classpath. I tried out a java program locally and added an unrelated record on the classpath and the same error happened.